### PR TITLE
change kbitem display in ticket (now with ajax)

### PR DESCRIPTION
**Pull request by orthagh**
Change Kb display in ticket.
Now use an ajax popover who display the corresponding kb entries.

Screenshot:
image

The feature is extended also to ticket task categories (new field).

### CODE REVIEW

File: ajax/getDropdownValue.php [hunk 1]
```diff
@@ -243,6 +243,8 @@
              ORDER BY $add_order `$table`.`completename`
              $LIMIT";
 
+   $datastoadd = array();
```
Comment (trasher) [inline ajax/getDropdownValue.php]: `data` is invariable.

File: inc/itilcategory.class.php [hunk 2]
```diff
@@ -234,11 +234,51 @@ function getLinks($withname=false) {
...
+         $kbitem = new KnowbaseItem;
+         $found_kbitem = $kbitem->find("`knowbaseitemcategories_id` = ".
+                                       $this->fields['knowbaseitemcategories_id']);
+
+         if (count($found_kbitem)) {
+            $ret.= "<div class='faqadd_block'>";
+            $ret.= "<label for='display_faq_chkbox'>";
+            $ret.= "<img src='".$CFG_GLPI["root_doc"]."/pics/faqadd.png' class='middle pointer'
+                      alt=\"$title\" title=\"$title\">";
+            $ret.= "</lebel>";
```
Comment (trasher) [inline inc/itilcategory.class.php]: Typo

File: inc/itilcategory.class.php [hunk 3]
```diff
@@ -234,11 +234,51 @@ function getLinks($withname=false) {
...
+            $ret.= "<div class='faqadd_entries'>";
+            if (count($found_kbitem) == 1) {
+               $ret.= "<div id='faqadd_block_content'>";
+               $ret.= $kbitem->showFull(['display' => false]);
+               $ret.= "</div>"; // #faqadd_block_content
+            } else {
+               $rand = mt_rand();
+               $ret.= Html::scriptBlock("
+                  var getKnowbaseItemAnswer = function() {
+                     console.log('getKnowbaseItemAnswer');
```
Comment (trasher) [inline inc/itilcategory.class.php]: Remove

File: install/mysql/glpi-9.2-empty.sql [hunk 4]
```diff
@@ -7040,6 +7040,7 @@ CREATE TABLE `glpi_taskcategories` (
   `entities_id` int(11) NOT NULL DEFAULT '0',
   `is_recursive` tinyint(1) NOT NULL DEFAULT '0',
   `taskcategories_id` int(11) NOT NULL DEFAULT '0',
+  `knowbaseitemcategories_id` int(11) NOT NULL DEFAULT '0',
```
Comment (trasher) [inline install/mysql/glpi-9.2-empty.sql:7043]: I do not understand why a field is added here; or maybe PR description is no correct.

**Review by trasher — CHANGES_REQUESTED** [review]

File: ajax/getDropdownValue.php (hunk 1 above)
Comment (orthagh) [reply ajax/getDropdownValue.php]: yeah, in fact, it's not a part of this pr, juste a residual bug .
I can remove this part and fix it in upstream

File: inc/itilcategory.class.php (hunk 2 above)
Comment (orthagh) [reply inc/itilcategory.class.php]: oh !

File: inc/itilcategory.class.php (hunk 3 above)
Comment (orthagh) [reply inc/itilcategory.class.php]: already done in local

File: install/mysql/glpi-9.2-empty.sql (hunk 4 above)
Comment (orthagh) [reply install/mysql/glpi-9.2-empty.sql:7043]: for recall: "WIP don't merge".
Some part are not finished

**Comment by orthagh** [comment]
@trasher work done and description updated, you can review.

**Comment by orthagh** [comment]
@trasher this branch is now rebased .

Could you check it ?

File: install/mysql/glpi-9.2-empty.sql (hunk 4 above)
Comment (trasher) [reply install/mysql/glpi-9.2-empty.sql:7043]: I still do not understand :)

Comment (trasher) [reply install/mysql/glpi-9.2-empty.sql:7043]: OK sounds like this permit to do something that was not possible before.. Ok for me.

**Review by trasher — APPROVED** [review]

**Pull request merged**